### ISSUE TITLE
NormalizingFlow base class updates

### ISSUE THREAD

**Issue by EthanMarx** [ML4GW/PE#43]
To generalize to arbitrary `embeddings`, the NormalizingFlow class will need some small tweaks:

The `__init__` should look as follows:
```python
def __init__(self, param_dim: int, n_ifos: int, strain_dim: int, num_flow_steps: int, embedding_net: torch.nn.Module = Flattener())
    self.param_dim = param_dim
    self.strain_dim = strain_dim
    self.n_ifos = n_ifos
    self.num_flow_steps = num_flow_steps
    self.embedding_net = embedding_net
```
`n_ifos` will be used to infer the context dimension as described in #42.

This change will require all sub classes (e.g maf, coupling) to take as their `shape`a `Tuple[int, int, int]` to account for the new `n_ifos` argument

**Comment by deepchatterjeeligo** [comment]
closed by #46
